src/battle-log.ts

```diff
@@ -412,6 +412,30 @@ class BattleLog {
 		let S = parseInt(hash.substr(0, 4), 16) % 50 + 40; // 40 to 89
 		let L = Math.floor(parseInt(hash.substr(8, 4), 16) % 20 + 30); // 30 to 49
 
+		let {R, G, B} = this.HSLToRGB(H, S, L);
+		let lum = R * R * R * 0.2126 + G * G * G * 0.7152 + B * B * B * 0.0722; // 0.013 (dark blue) to 0.737 (yellow)
+
+		let HLmod = (lum - 0.2) * -150; // -80 (yellow) to 28 (dark blue)
+		if (HLmod > 18) HLmod = (HLmod - 18) * 2.5;
+		else if (HLmod < 0) HLmod = (HLmod - 0) / 3;
+		else HLmod = 0;
+		// let mod = ';border-right: ' + Math.abs(HLmod) + 'px solid ' + (HLmod > 0 ? 'red' : '#0088FF');
+		let Hdist = Math.min(Math.abs(180 - H), Math.abs(240 - H));
+		if (Hdist < 15) {
+			HLmod += (15 - Hdist) / 3;
+		}
+
+		L += HLmod;
```

Comment:
You don't run `HSLtoRBG` again after modding `L` here.

Reply:
Fuck, I even knew I had to call it twice but then didn't do it. *sigh*. TY.